src/openms/include/OpenMS/ANALYSIS/ID/PrecursorPurity.h

```diff
@@ -78,7 +78,7 @@ namespace OpenMS
      * @param precursor_mass_tolerance The precursor tolerance. Is used for determining the targeted peak and deisotoping.
      * @param precursor_mass_tolerance_unit_ppm The unit of the precursor tolerance
     */
-    static std::vector<PurityScores> computePrecursorPurities(const PeakMap& spectra, double precursor_mass_tolerance, bool precursor_mass_tolerance_unit_ppm);
+    static std::map<String, PurityScores> computePrecursorPurities(const PeakMap& spectra, double precursor_mass_tolerance, bool precursor_mass_tolerance_unit_ppm);
```

Comment:
instead of switching to a map (slow), could you also preallocate  the vector outside the parallel region and use `operator[]` to access for each thread. This means that indices for spectra which are skipped over remain untouched. If this is not a problem, then you can get rid of the `critical` section (which might be broken on Windows...)

Reply:
@cbielow one of the issues I want to avoid with this, is the dependency between the order of the spectra in the peak map and this list The peak map is processed further in my tool and some of the spectra get filtered out. Using a map is much simpler and less error prone, than what I did before: saving the indices of the removed spectra and then removing the same ones from this list. Also this map only has the same number of items as there are MS2 spectra in the peak map and won't become a performance bottleneck anytime soon. The parallelization of PrecursorPurity is just a bonus that I added, because it was simple to do now. Using a map makes handling this class much easier and made the parallelization of my spectrum deisotoping and filtering also simpler.